### PR TITLE
Refactor argument passing and command structure, fixes #32

### LINKED ISSUE #32: Wildcard support to upload last modified file with that name
At the moment we can only upload files we know the exact name, so adding a wildcard could be used for example to upload a build file right after the compile build command finishes 
`gdrive upload fastboot*`

### PR DESCRIPTION
**Pull request by jrcamelo**
Refactors the manual argument parsing with ArgParse  
Changes commands for easier argument parsing configuration and a clearer OO structure
Remade #33 as to be able to use #32 with ArgParse
Keeps the unmerged change made previously to remedy #17

### CODE REVIEW

**Review by psteiger — APPROVED** [review]
Looks good to me! Congrats

**Comment by jrcamelo** [comment]
ArgParser documentation:
https://docs.python.org/3/howto/argparse.html

Class structure:
**- GoogleCredentials**: Manages Google Drive credentials and is used in GoogleService to get API
**- GoogleService**: Encapsulates Google Drive API while keeping some API-side configurations
**- Command**: Base class for commands. Commands configure their own argument parsers
**- Download**: Can download through filename and ID, or the last modified file
**- Upload**: Can upload a specific file, accepts wildcard, and can find the last modified file
**- List**: Can search for files or browse all files
**- CommandParser**: Builds the argument parsers for the commands and returns a command if valid

**Review by viniciusalbuquerque — APPROVED** [review]

File: gdrive [hunk 1]
```diff
@@ -18,271 +17,412 @@ from googleapiclient.http import MediaIoBaseDownload
...
+            print("Command is not valid")
+            return
+        command = self.build_command()
+        if not command.is_service_started():
+            print("Google Drive Service failed to start")
+            return
+        command.execute()
+    
+    def is_command_valid(self):
+        return self.args or self.args.command
```
Comment (jrcamelo) [inline gdrive:422]: Small mistake found where an OR should be an AND
It is but a safety measure for code mistakes, so not relevant at the moment
Will fix it in a later commit

Comment (jrcamelo) [reply gdrive:422]: For the record, argparse already deals with user command mistakes

> usage: gdrive [-h] {download,upload,list} ...
> gdrive: error: invalid choice: 'invalid_command' (choose from 'download', 'upload', 'list')

**Pull request merged**